File: LinkedList/206. Reverse Linked List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* reverseList(struct ListNode* head){
    if(head==NULL){return head;}
    if(head->next==NULL){
        return head;
    }
    
    struct ListNode* prev=head;
    struct ListNode* Next;
    struct ListNode* Now=head->next;
    
    head->next=NULL;
    
    if(Now->next==NULL){//case 2
        Now->next=prev;
        return Now;
    }
    while(Now->next!=NULL){
        Next=Now->next;
        Now->next=prev;
        prev=Now;
        Now=Next;
    }
    Now->next=prev;
    return Now;
}

File: LinkedList/206. Reverse Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    Stack<ListNode> x = new Stack<ListNode>();
    public ListNode reverseList(ListNode head) {
        if(head==null){
            return null;
        }
        while(head.next!=null){
            x.push(head);
            head = head.next;
        }
        ListNode Next = head;
        while(!x.empty()){
            Next.next = x.pop();
            Next = Next.next;
        }
        Next.next = null;
        return head;
    }
}